### DISCUSSION TITLE
Custom apex domain 404 error

### DISCUSSION THREAD

**Discussion by akruse10** [community/community#132013] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello! I have a custom apex domain and I am trying to host my github pages site on it but I am getting a 404 'there is not a github pages site here' when I visit the custom domain. 
![image](https://github.com/user-attachments/assets/1c84b319-031f-4663-8cf0-28882ed9b9c3)
Here are the DNS records for the custom domain, samkruse.com. 

Here is my repo, where I have a gh-pages branch which has a CNAME file with the content samkruse.com
https://github.com/akruse10/samportfolio/tree/gh-pages

I had the custom domain working fine at the beginning of the project, but after pushing a bunch of new builds and trying to get it to work again, I am stuck. Not sure what I would have changed. I am using create-react-app which I know is deprecated and could be the source of some issues. 

I believe this is all I need to have the custom domain working, but I am not sure where to go from here to resolve the 404 error. I have done a lot of googling and documentation reading but have gotten nowhere so I really appreciate any help or troubleshooting advice. Thanks in advance!

**Comment by knarrff** [comment]
Your pages are published, but not to your custom domain: https://akruse10.github.io/samportfolio/, which means github doesn't pick up / know about your domain.

Can you post a screenshot of your github pages configuration: https://github.com/akruse10/samportfolio/settings/pages ?

**Reply by akruse10** [reply]
![image](https://github.com/user-attachments/assets/74a15e20-851c-44a1-a8a6-f87d8944d245)
I am confused what you mean by the first part. I believe my pages are published to that domain? They show up properly? I must not understand what published means. Can you explain?

**Reply by akruse10** [reply]
when I try to use the custom domain section of the repos settings/pages, it says that domain is already in use. I assume this is by me since I already have a CNAME with that domain? or is someone else using my domain in a repo?

**Reply by knarrff** [reply]
I guess the question now is what is using that name - it could be https://github.com/akruse10/akruse10.github.io : it does publish to https://akruse10.github.io, while samportfolio publishes to https://akruse10.github.io/samportfolio/.

Also, which custom domain did you try to enter in the settings, samkruse.com or www.samkruse.com?

My best guess as to what happens is: in samportfolio you try to add www.samrkuse.com as domain. Github looks at the DNS record and sees a CNAME to akruse10.github.io. It then looks up which repo publishes there and sees that's https://github.com/akruse10/akruse10.github.io - and then gives an error since that is not the repo you wanted to set the domain to.

**Reply by akruse10** [reply]
I originally tried samkruse.com but both samkruse.com and www.samkruse.com return the error "that domain is already taken" on github. I do see now that the repo named akruse.github.io could be the issue. I shall delete it and see if that fixes anything.

**Reply by akruse10** [reply]
It did not immediately fix the problem, but I will wait and check again in a bit. I also wonder if I should try deleting the CNAME file from my repo and use the custom domain section in settings/pages to add it? Would that do anything differently or does it simply add a CNAME file to my repo?

**Reply by akruse10** [reply]
Do you think changing the CNAME in my repo from samkruse.com to www.samkruse.com would affect anything? Can you possibly explain the difference to me? By the way, thank you IMMENSELY for all you help so far. I have been struggling a lot to find help with this part of the project and I am so close to finishing. It's a portfolio for my brother and he is very eager to show it off to our family and friends. :)

**Reply by knarrff** [reply]
I think having the CNAME file there at least should not hurt. Depending on how exactly you publish the pages you might not need it (or so I read). However, it should also not hurt to remove it (from git) and then try to enter the domain in the web frontend. I am afraid, apart from removing and re-adding information and leaving enough time in between I don't know what else to do.

**Reply by knarrff** [reply]
Changing the CNAME in my repo from samkruse.com to www.samkruse.com might make it work, although since the web gui also gives an error for that domain, I am not so sure.

The question here is how you want the site to be reached: with or without www.

If all else fails, I think you should contact support. If not for anything else, note also the warning here:

https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages

I am not saying that this is the case for you necessarily, but maybe it is:


[Securing the custom domain for your GitHub Pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages#securing-the-custom-domain-for-your-github-pages-site)

If your GitHub Pages site is disabled but has a custom domain set up, it is at risk of a domain takeover. Having a custom domain configured with your DNS provider while your site is disabled could result in someone else hosting a site on one of your subdomains.

Verifying your custom domain prevents other GitHub users from using your domain with their repositories. If your domain is not verified, and your GitHub Pages site is disabled, you should immediately update or remove your DNS records with your DNS provider. For more information, see "[Verifying your custom domain for GitHub Pages](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages)" and "[Managing a custom domain for your GitHub Pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site)."

**Reply by akruse10** [reply]
Thanks for the help. I tried verifying the domain and still get the same error which is weird. I submitted a help ticket and will come back here if I end up getting an answer!

**Reply by LiteBrite82** [reply]
@akruse10 
Who is your DNS host?

**Reply by akruse10** [reply]
I am using GoDaddy. I had the custom domain working with the github pages site previously in the project with the same DNS records in GoDaddy so I am hesitant to touch those, I suspect the problem is on githubs side or in my code.

**Reply by akruse10** [reply]
I do have the samkruse.com domain verified now with github but I am getting the same error when trying to use the web gui to enter a custom domain. I don't understand how the domain can be taken and how it could be telling me to verify the domain if I am the one it is taken by and I have the domain verified already!! Ugh!!!

**Reply by LiteBrite82** [reply]
For your Pages Custom Domain settings: 
![Screenshot 2024-07-31 at 10 09 04 AM](https://github.com/user-attachments/assets/980b8cab-cb97-4005-aba6-e34b91628c91)

Are you using `www.samkruse.com` or `samkruse.com`

**Reply by akruse10** [reply]
i have tried both and it says both are taken

**Comment by LiteBrite82** [comment, marked as answer]
@akruse10 
I checked with the Devs, your custom domain has been taken over by someone else. You need to take it back by going through the [verification process](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages)

**Reply by LiteBrite82** [reply]
@ me when you are done with the verification process so I can help expedite

**Reply by akruse10** [reply]
oh jeez. ok i have done the verification process earlier, i believe its complete. does this look right?
![image](https://github.com/user-attachments/assets/d6567d8b-c83b-46a6-af82-89ca3cda4928)
@LiteBrite82

**Reply by akruse10** [reply]
thank you for the help btw.

**Reply by akruse10** [reply]
do i need to try and verify www.samkruse.com as well? i am not sure of the difference

**Reply by akruse10** [reply]
I understood the documentation on the verification process up until the "release" part. I read that once I verify its supposed to "release" my domain but I haven't seen how to check if it's released or been able to find much else about the release process? I am not sure if my domain has been released or not. I am thinking not

**Reply by LiteBrite82** [reply]
Just checked, it is released

**Reply by akruse10** [reply]
does that mean I can now claim it via the web gui?

**Reply by LiteBrite82** [reply]
Yes, ASAP

**Reply by akruse10** [reply]
YAY it worked! samkruse.com still has a 404 error but it worked to save the CNAME

**Reply by LiteBrite82** [reply]
🤞

**Reply by akruse10** [reply]
Ill give it some time

**Reply by akruse10** [reply]
Thanks so much!!!!!

**Reply by LiteBrite82** [reply]
thats progress!!!

**Reply by akruse10** [reply]
Yes for sure! this is the most progress I've made in weeks so I'm very very happy

**Reply by akruse10** [reply]
I will keep checking and let you know if the site populates

**Reply by akruse10** [reply]
it says DNS check in progress under the custom domain section of the web gui so thats good

**Reply by LiteBrite82** [reply]
Sweet! I'm not getting a 404 anymore!!! 
![Screenshot 2024-07-31 at 12 39 26 PM](https://github.com/user-attachments/assets/b838b38d-88fb-49cb-bb7b-c51ecb421864)

Looks like we have to solve a build issue now. Thats a whole different can of worms; using Create-React-App will make it difficult.

**Reply by akruse10** [reply]
Ok this is not good. I'm confused because I had it working fine on akruse10.github.io/samportfolio

**Reply by akruse10** [reply]
Progress tho!

**Reply by LiteBrite82** [reply]
Transitioning to Vite is relatively painless. You just have to rename any TypeScript files with `.js` to `.jsx`

**Reply by akruse10** [reply]
any words of wisdom? I'm not sure where to go from here. might need to make a new post now lol

**Reply by akruse10** [reply]
ok that is a good plan

**Reply by akruse10** [reply]
I didn't realize this would cause the old link, akruse10.github.io/samportfolio, to snap to samkruse.com and now break. If I remove the domain so that I can get the old link back (because I'm using it on my resume) am I at risk of another takeover or am I fine now since I have it verified?

**Reply by LiteBrite82** [reply]
Yeah, I would start a new post in `Programming Help`; could be faster than 'Pages'. 

There is a lot of crossover between `Pages` and `Programming Help` on this topic; you can find past discussions in both.

**Reply by LiteBrite82** [reply]
also, see this: https://github.com/orgs/community/discussions/128670

**Reply by akruse10** [reply]
thank you @LiteBrite82 !!! Do you know the answer to my question above?

**Reply by LiteBrite82** [reply]
Good question...... I dont have all of the information I need to accurately answer that question; I dont know what is causing your page to break. 

One thing you could experiment with, would be to create a new repo and copy your project into it, then try redeploying from there without using your custom domain.

**Reply by LiteBrite82** [reply]
Also, you need to run a new build of your CRA and redeploy. 
- You shouldn't be making any manual changes to the code in the Gh-Pages branch; that should only contain code that has gone through the build process

**Reply by LiteBrite82** [reply]
check out this post on how to automate the react deployment from your package.json script: https://blog.logrocket.com/deploying-react-apps-github-pages/

Then all you have to do, anytime you make changes to your code in the `main` branch, is type "npm run deploy" in your console. It will automatically create a build and push it to the Gh-Pages branch and deploy.

**Reply by akruse10** [reply]
@LiteBrite82 I fixed it! check out samkruse.com!!!! Thanks for all your help. just had to change the homepage value in package.json. woohooo!!!!!

**Reply by LiteBrite82** [reply]
Its so good!!! Fantastic photography!!!! lets go Seattle! I'm excited for Seafair this weekend!!!

**Comment by oussemamansouri** [comment]
Can you explain more ?

**Reply by LiteBrite82** [reply]
@oussemamansouri 

Sure, can you be more specific.

**Comment by kalilbuti1234-pixel** [comment]
انشاء تطبيق سكننار 1.0

**Comment by kalilbuti1234-pixel** [comment]
[_._._.xlsx](https://github.com/user-attachments/files/31883953/_._._.xlsx)
[قاعدة البيانات — ويكيبيديا.pdf](https://github.com/user-attachments/files/31883958/default.pdf)

**Comment by kalilbuti1234-pixel** [comment]
ربط  تطبيق  سكنار  1.0
